### PR TITLE
Remove macos conditional code.

### PR DESCRIPTION
**Pull request by porcuquine**

### CODE REVIEW

File: src/cl.rs [hunk 1]
```diff
@@ -110,19 +108,26 @@ fn create_futhark_context(device: bindings::cl_device_id) -> ClResult<FutharkCon
...
-        map.insert(bus_id, Arc::new(Mutex::new(context)));
+
+    let key = selector.get_key();
+    info!("device key: {}", key);
+    if !map.contains_key(&key) {
+        if let Some(device) = selector.get_device() {
+            info!("device: {:?}", device);
+            let cl_device_id = unsafe {
+                std::mem::transmute::<cl_device_id, bindings::cl_device_id>(device.cl_device_id())
+            };
```
Comment (dignifiedquire) [inline src/cl.rs:119]: this transmute is actually not needed

```rust
let cl_device_id = device.cl_device_id() as _;
```
will just work fine

File: src/cl.rs [hunk 2]
```diff
@@ -110,19 +108,26 @@ fn create_futhark_context(device: bindings::cl_device_id) -> ClResult<FutharkCon
...
+            let context = create_futhark_context(cl_device_id)?;
+            map.insert(key.clone(), Arc::new(Mutex::new(context)));
+        } else {
+            unimplemented!();
```
Comment (dignifiedquire) [inline src/cl.rs:123]: shouldn't this error out?

File: src/cl.rs (hunk 1 above)
Comment (porcuquine) [reply src/cl.rs:119]: Oh, great.

File: src/cl.rs (hunk 2 above)
Comment (dignifiedquire) [reply src/cl.rs:123]: In the sense of not panicing

Comment (porcuquine) [reply src/cl.rs]: Yeah, fixed now. That was a placeholder.

**Review by dignifiedquire — APPROVED** [review]

**Pull request merged**